Products/CMFPlone/image_scales/adapters.py

```diff
@@ -0,0 +1,158 @@
+from zope.globalrequest import getRequest
+from plone.namedfile.interfaces import INamedImageField
+from Acquisition import aq_inner
+from plone.dexterity.interfaces import IDexterityContent
+from plone.dexterity.utils import iterSchemata
+from zope.component import adapter
+from zope.component import getMultiAdapter
+from zope.component import queryMultiAdapter
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.schema import getFields
+from plone.registry.interfaces import IRegistry
+from zope.component import getUtility
+from zope.security.interfaces import IPermission
+from Products.CMFPlone.image_scales.interfaces import IImageScalesAdapter
+from Products.CMFPlone.image_scales.interfaces import IImageScalesFieldAdapter
+from Products.CMFPlone.interfaces import IImagingSchema
+
+
+@implementer(IImageScalesAdapter)
+@adapter(IDexterityContent, Interface)
+class ImageScales:
+    def __init__(self, context, request):
+        self.context = context
+        self.request = request
+
+    def __call__(self):
+        obj = aq_inner(self.context)
+        res = {}
+
+        for schema in iterSchemata(self.context):
+            for name, field in getFields(schema).items():
+                # serialize the field
+                serializer = queryMultiAdapter(
+                    (field, obj, self.request), IImageScalesFieldAdapter
+                )
+                if serializer:
+                    scales = serializer()
+                    if scales:
+                        res[name] = scales
+        return res
+
+
+@implementer(IImageScalesFieldAdapter)
+@adapter(INamedImageField, IDexterityContent, Interface)
+class ImageFieldScales:
+    def __init__(self, field, context, request):
+        self.context = context
+        self.request = request
+        self.field = field
+
+    def __call__(self):
+        image = self.field.get(self.context)
+        if not image:
+            return
+
+        width, height = image.getImageSize()
+
+        url = self.get_original_image_url(self.field.__name__, width, height)
+
+        scales = self.get_scales(self.field, width, height)
+        return {
+            "filename": image.filename,
+            "content-type": image.contentType,
+            "size": image.getSize(),
+            "download": url,
+            "width": width,
+            "height": height,
+            "scales": scales,
+        }
```

Comment:
We want to have a list here (think of multi-image fields, like a relation to images), right?

Reply:
Yes, sorry. Now it returns a list.
Should i add additional adapters in base implementation or not? Because multi-image fields are related to custom implementations.

Reply:
We could do so for a RelationList, not sure if we want to have it in Core?